addons/account/static/src/js/section_and_note_backend.js

```diff
@@ -0,0 +1,64 @@
+/*
+    The goal of this file is to contain JS hacks related to allowing
+    section and note on sale order and invoice.
+*/
+odoo.define('account.list_renderer_section_and_note', function (require) {
+
+"use strict";
+var ListRenderer = require('web.ListRenderer');
+var fieldRegistry = require('web.field_registry');
+var FieldOne2Many = require('web.relational_fields').FieldOne2Many;
+
+var SectionAndNoteListRenderer = ListRenderer.extend({
+    _renderBodyCell: function (record, node, index, options) {
+        var $cell = this._super.apply(this, arguments);
+
+
+        var isSection = record.data['display_type'] === 'line_section';
+        var isNote = record.data['display_type'] === 'line_note';
+
+        if (isSection || isNote) {
+            if ($cell.hasClass('o_handle_cell')) {
+                return $cell;
+            } else if ($cell.hasClass('product_name')) {
+                // TODO SEB: for section, make it only input text (not textarea) on editable mode
+
+                var nbrColumns = this._getNumberOfCols();
+                if (this.handleField) {
+                    nbrColumns--;
+                }
+                if (this.addTrashIcon) {
+                    nbrColumns--;
+                }
+                $cell.attr('colspan', nbrColumns);
+            } else {
+                return $cell.addClass('o_hidden');
+            }
+        }
+
+        return $cell;
+    },
+    // We add the is_{display_type} class to allow custom behaviour both in JS and CSS.
+    _renderRow: function (record, index) {
+        var $row = this._super.apply(this, arguments);
+
+        if (record.data['display_type']) {
+            $row.addClass('o_is_' + record.data['display_type']);
+        }
+
+        return $row;
+    },
+});
+
+var SectionAndNoteFieldOne2Many = FieldOne2Many.extend({
```

Comment:
explain why we create a widget: this is the cleanest way to do it, to be sure this custom code will only impact selected fields having the widget and not be ran for any other default one